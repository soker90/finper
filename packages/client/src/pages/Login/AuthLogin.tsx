import { useState } from 'react'

import {
  Button,
  FormHelperText,
  Grid,
  IconButton,
  InputAdornment,
  InputLabel,
  OutlinedInput,
  Stack
} from '@mui/material'
import { EyeOutlined, EyeInvisibleOutlined } from '@ant-design/icons'
import { useForm } from 'react-hook-form'
import { SendLoginParams, useLogin } from './hooks'

const AuthLogin = () => {
  const [showPassword, setShowPassword] = useState(false)
  const { register, handleSubmit, formState: { errors } } = useForm({
    defaultValues: {
      username: '',
      password: ''
    }
  })
  const { sendLogin, error, loading } = useLogin()

  const onSubmit = handleSubmit(data => {
    sendLogin(data as SendLoginParams)
  })

  const handleClickShowPassword = () => {
    setShowPassword(state => !state)
  }

  return (
        <form onSubmit={onSubmit}>
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <Stack spacing={1}>
                        <InputLabel htmlFor="username">Usuario</InputLabel>
                        <OutlinedInput
                            id='username'
                            placeholder="Introduce tu nombre de usuario"
                            fullWidth
                            error={!!errors.username}
                            {...register('username', { required: true, minLength: 3 })}
                        />
                        {errors.username && (
                            <FormHelperText error>
                                Introduce un nombre de usuario válido
                            </FormHelperText>
                        )}
                    </Stack>
                </Grid>
                <Grid item xs={12}>
                    <Stack spacing={1}>
                        <InputLabel htmlFor="password">Contraseña</InputLabel>
                        <OutlinedInput
                            id='password'
                            fullWidth
                            error={!!errors.password}
                            type={showPassword ? 'text' : 'password'}
                            endAdornment={
                                <InputAdornment position="end">
                                    <IconButton
                                        aria-label="toggle password visibility"
                                        onClick={handleClickShowPassword}
                                        edge="end"
                                        size="large"
                                    >
                                        {showPassword ? <EyeOutlined /> : <EyeInvisibleOutlined />}
                                    </IconButton>
                                </InputAdornment>
                            }
                            placeholder="Introduce la contraseña"
                            {...register('password', { required: true, minLength: 5 })}
                        />
                        {errors.password && (
                            <FormHelperText error>
                                Introduce una contraseña válida
                            </FormHelperText>
                        )}
                    </Stack>
                </Grid>

                {error && (
                    <Grid item xs={12}>
                        <FormHelperText error>{error}</FormHelperText>
                    </Grid>
                )}

                <Grid item xs={12}>
                    <Button
                        disableElevation
                        disabled={loading}
                        fullWidth
                        size="large"
                        type="submit"
                        variant="contained"
                        color="primary"
                    >
                        {loading ? 'Iniciando...' : 'Iniciar sesión'}
                    </Button>
                </Grid>
            </Grid>
        </form>

  )
}

export default AuthLogin
