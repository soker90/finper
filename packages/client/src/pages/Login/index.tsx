import { Grid, Stack, Typography } from '@mui/material'

import AuthLogin from './AuthLogin'

const Login = () => (
    <Grid container spacing={3}>
        <Grid item xs={12}>
            <Stack direction="row" justifyContent="space-between" alignItems="baseline" sx={{ mb: { xs: -0.5, sm: 0.5 } }}>
                <Typography variant="h3">Login</Typography>
            </Stack>
        </Grid>
        <Grid item xs={12}>
            <AuthLogin/>
        </Grid>
    </Grid>

)

export default Login
