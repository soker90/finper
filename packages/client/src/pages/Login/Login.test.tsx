import { describe, it } from 'vitest'
import Login from './index'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

const user = userEvent.setup()

describe('Login', () => {
  it.skip('should redirect to dashboard', async () => {
    const { getByPlaceholderText } = render(<Login />)
    const inputUsername = getByPlaceholderText('Introduce tu nombre de usuario')
    const inputPassword = getByPlaceholderText('Introduce tu contrase√±a')

    await user.type(inputUsername, 'test')
    await user.type(inputPassword, 'password')
  })
})
