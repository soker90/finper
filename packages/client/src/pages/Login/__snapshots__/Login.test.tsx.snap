// Vitest Snapshot v1

exports[`Login > snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-6dalm9-MuiGrid-root"
    >
      <div
        class="css-1h1s0jv-MuiStack-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-16iw3b8-MuiTypography-root"
        >
          Inicio de sesión
        </h3>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-6dalm9-MuiGrid-root"
    >
      <form>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-6dalm9-MuiGrid-root"
          >
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-1sbd9j1-MuiFormLabel-root-MuiInputLabel-root"
                for="username"
              >
                Usuario
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-1a7pcy6-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1ci3ki6-MuiInputBase-input-MuiOutlinedInput-input"
                  defaultvalue=""
                  id="username"
                  placeholder="Introduce tu nombre de usuario"
                  type="text"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-6wp6pb-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-6dalm9-MuiGrid-root"
          >
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-1sbd9j1-MuiFormLabel-root-MuiInputLabel-root"
                for="password"
              >
                Contraseña
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedEnd css-1trfusx-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1i5uhia-MuiInputBase-input-MuiOutlinedInput-input"
                  defaultvalue=""
                  id="password"
                  placeholder="Introduce la contraseña"
                  type="password"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeLarge css-1a13r4v-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      aria-label="eye-invisible"
                      class="anticon anticon-eye-invisible"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="eye-invisible"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                        />
                        <path
                          d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-6wp6pb-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-6dalm9-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1rke8zj-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Iniciar sesión
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
